tests/test_parser.py

```diff
@@ -748,3 +748,18 @@ def test_label_invalid(self, dfparser, value):
         ])
         with pytest.raises(Exception):
             dfparser.labels
+
+    def test_remove_whitespace(self, tmpdir):
+        """
+        Verify keys are parsed correctly even if there is no final newline.
+
+        """
+        with open(os.path.join(str(tmpdir), 'Dockerfile'), 'wb') as fp:
```

Comment:
Opening file in binary mode causes python3 to require bytes to be written to it. Just 'w' should do here since you are writing a text file.

Reply:
Fixed.